js/components/conceptsetInclusionCount/conceptsetInclusionCount.js

```diff
@@ -22,7 +22,7 @@ define([
 			this.conceptSetExpression = params.conceptSetExpression || ko.observableArray();
 
 			this.getInclusionCount = this.getInclusionCount.bind(this);
-			ko.computed(() => ko.toJSON(this.conceptSetExpression)).subscribe(this.getInclusionCount);
+			this.conceptSetExpressionSub = ko.pureComputed(() => ko.toJSON(this.conceptSetExpression)).subscribe(this.getInclusionCount);
```

Comment:
What would you say about adding a rate-limit to this computed, so that the subscription is only fired after a little bit of quiet?  This way when someone is rapidly adding/removing concepts or they switch around the inclusion checkboxes, we only signal the sub after a 1 second delay?  It would be done like so:

```
// Shorthand: Specify just a timeout in milliseconds
someObservableOrComputed.extend({ rateLimit: 1000 });
```

From: https://knockoutjs.com/documentation/rateLimit-observable.html

Reply:
What benefits will it give us?

Reply:
The get inclusion count (if i understand it correctly) is a pretty heavy DB operation.  Maybe if there was caching in place, this wouldn't be much of a problem.  But, the normal activity of managing a concept set is to set up a set of high-level concepts, look at the included concepts, select the ones you want to remove, then click on 'exclude', then for some of them you want to exclude their descendants, so you exclude those...  

Here's a hypothetical sequence of events for the above example without rate limiting:
1. Search for term
2. Select a high level term.
3. Go to concept set editor {call: getInclusionCount()}
4. Go to included concepts, filter on some term (like pregnancy or 'history of')
5. select the first concept you want to exclude  {call: getInclusionCount()}
6. select a second concept to exclude  {call: getInclusionCount()}
7. select a third concept to exclude  {call: getInclusionCount()}
8. switch back to main concept set. click the first concept's exclude checkbox  {call: getInclusionCount()}
9. select next one to exclude  {call: getInclusionCount()}
10. select next one to exclude  {call: getInclusionCount()}
11. select first to include descendants  {call: getInclusionCount()}
12. select next one to include descendants  {call: getInclusionCount()}

Conversely, if we rate-limit the notification to the subscription you get this
1. Search for term
2. Select a high level term.
3. Go to concept set editor {call: getInclusionCount()}
4. Go to included concepts, filter on some term (like pregnancy or 'history of')
5. select the first concept you want to exclude  {call: getInclusionCount()}
6. select a second concept to exclude  {no call if selection is made within the rateLimit}
7. select a third concept to exclude  {no call if selection is made within the rateLimit}
8. switch back to main concept set.  {call: getInclusionCount() since the last selection will now notify subs}
9. select first concept to  exclude exclude  {rateLimit begins)
10. select next one to exclude  {no call if selection is made within the rateLimit}
11. select next one to exclude  {no call if selection is made within the rateLimit}
12. select first to include descendants  {no call if selection is made within the rateLimit}
13. select next one to include descendants  {call: getInclusionCount() since the last selection will now notify subs}

Every call to getInclusionCount is a relatively expensive query to the concept_ancestor table where we need to include/exclude descendants and it's not a trivial thing.  Any way we can limit the interaction to the database would be ideal.  I think it would be hard to cache the results of the concept set expression count because the permutations of the possible selections within a concept set leads it to be something that is not easily cached.